source/blog/_posts/2016-01-26-shopware-environment.md

```diff
@@ -0,0 +1,143 @@
+---
+draft: true
+title: Shopware Environment
+tags:
+    - environment
+    - htaccess
+    - vhost
+
+categories:
+- dev
+github_link: blog/_posts/2016-01-26-shopware-environment.md
+
+authors: [bc]
+---
+
+Shopware is able to handle multiple environments using the `SHOPWARE_ENV` environment variable.
+
+## Setting the environment
+
+### Apache
+
+You can set the environment for your local installation using the `.htaccess`:
+
+```
+SetEnv SHOPWARE_ENV dev
+```
+
+This also can be done conditionally for a specifix Host or Subshop:
+
+```
+SetEnvIf Host "dev.shopware.in" SHOPWARE_ENV=dev
+```
+
+On my local development machine I configured it in the apache vhost configuration for all installations:
+
+```
+<VirtualHost *:80>
+    DocumentRoot "/home/bcremer/www/"
+    SetEnv SHOPWARE_ENV dev
+</VirtualHost>
+```
+
+### Nginx
+When using the [shopware-with-nginx](https://github.com/bcremer/shopware-with-nginx) configuration:
+
+
+```
+set $shopware_env 'production';
+```
+
+### Ant
+
+```bash
+$ SHOPWARE_ENV=dev ant build-unit
+```
+
+### Shopware Console
+
+This also works for the [Shopware Console Commands](/developers-guide/shopware-5-cli-commands/):
+
+
+```bash
+$ SHOPWARE_ENV=dev ./bin/console sw:generate:attributes
+```
+
+Alternativly you can also use the `--env` option that exists for every console command:
+
+```bash
+$ ./bin/console sw:generate:attributes --env=dev
+```
+
+### Export Environment globally
+`$HOME/.profile`
+
+```
+export SHOPWARE_ENV=dev
+```
+
+## How the enviroment is handled
+
+This environment variable is picked by by [`shopware.php`](https://github.com/shopware/shopware/blob/5.2/shopware.php#L92) and [`bin/console`](https://github.com/shopware/shopware/blob/5.2/bin/console#L34) and then propagated to the kernel:
+
+```php
+$environment = getenv('SHOPWARE_ENV') ?: 'production';
+$kernel = new Kernel($environment);
+```
+
+As you can see, if no environment is given, `production` is used as the default.
+
+
+## Generated files
+
+The environment taken into account for the cache directories and log files.
+Running Shopware each with `dev`, `production` and `testing` environment will result in the following directory tree:
```

Comment:
Running Shopware once with each of the environments (`dev`, `production` and `testing`) will...